server/lib/scene/scene.addScene.js

```diff
@@ -26,14 +27,23 @@ const nodeScheduleDaysOfWeek = {
  *  selector: 'test'
  * });
  */
-function addScene(sceneRaw) {
+async function addScene(sceneRaw) {
   // deep clone the scene so that we don't modify the same object which will be returned to the client
   const scene = cloneDeep(sceneRaw);
   // first, if the scene actually exist, we cancel all triggers
   this.cancelTriggers(scene.selector);
   // Foreach triggger, we schedule jobs for triggers that need to be scheduled
   if (scene.triggers) {
+    const houses = await this.house.get();
     scene.triggers.forEach((trigger) => {
+      let times;
+      if (undefined !== trigger.house) {
+        const house = houses.find((findHouse) => findHouse.selector === trigger.house);
+        if (house !== undefined) {
+          times = sunCalc.getTimes(new Date(), house.latitude, house.longitude);
```

Comment:
I don't think it works to calculate the sunrise/sunset time here in the addScene function.

Sunrise/sunset time changes everyday, so the sunCalc.getTime() should be calculated on the start of each day (at 00:00), and several events should be scheduled: 

- Sunrise for this day for each house
- Sunset for this day for each house

What I would do: 

- Create a recurrence rule (enforced), that runs at 00:00 every day, and schedule events for sunrise/sunset
- Foreach house, in this task, schedule sunrise/sunset, no matter if a scene exists with this house

Reply:
@Pierre-Gilles  : 
Thanks for your return 
You are right, it's more simple right now :)

I have made the change. Did I understand what you explained to me ?

Reply:
Yes, it looks way better now! :)

Reply:
Once you think that your PR is ready to be merged, that the code coverage is good, you can remove the draft on the PR and I'll do a full review of the PR :)

Reply:
@Pierre-Gilles : I'm sorry, I don't find the good way to increase my code coverage. Do you have any idea ?
I can't test the result of schedule.scheduleJob because this function return null is the date has already passed.

Reply:
<img width="945" alt="Screenshot 2021-01-15 at 09 58 33" src="https://user-images.githubusercontent.com/7365207/104704076-7a712880-5718-11eb-8e42-07c70f875fa2.png">

It's not the schedule function that you need to test, is the callback that's missing.

So you need to simulate a sunset and a sunrise in your code (for example by scheduling a sunrise/sunset in 100ms and see what happens), it'll trigger the callback

It's very important that you test this part of your code, because this is the "core feature" :)

Reply:
I have found a solution to test that. It's good now :)